### DISCUSSION TITLE
🚀 Feature Request: D1 restore sqlite binary files

### DISCUSSION THREAD

**Discussion by rozenmd** [cloudflare/workers-sdk#13546] [Feature Requests]
### Describe the solution

- Given that `wrangler d1 backup download ...` sends me a SQLite binary file, I would expect to be able to immediately use that file to restore my DB (without relying on existing backups stored on Cloudflare)
- This'll require running a `.dump` on the binary file and executing the SQL output (as we can't trust user-provided binaries)

**Comment by YankeeTube** [comment]
I agree with this feature proposal.

Currently, if you want to use a .sql file to insert or overwrite large-scale data, you have to delete it and put it back in, but if the data is vast, it fails.

Therefore, there is an inconvenience of dividing the data by 20,000 pieces and continuing to request insertion.

**Comment by penalosa** [comment]
@rozenmd is this still being considered?

**Comment by rozenmd** [comment]
@penalosa yep
